subt/dummy_darpa_server.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+"""
+  webserver with connection to serial controller
+  Usage:
+     ./web2.py [<COM port>|dummy]
+"""
+
+#import string,cgi,time
+import os
+import sys
+import csv
+import math
+import json
+from collections import defaultdict
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from mimetypes import guess_type
+
+
+def dist3d(xyz, xyz2):
+    return math.sqrt(sum([(a-b)**2 for a, b in zip(xyz, xyz2)]))
+
+
+class GameLogic:
+    def __init__(self, filename):
+        self.score = 0
+        self.artf = defaultdict(list)
+        with open(filename) as csvfile:
+            reader = csv.reader(csvfile)
+            for raw in reader:
+                if 'artf' in raw:
+                    continue
+                self.artf[raw[0]].append(tuple([float(x) for x in raw[1:]]))
+
+    def report_artf(self, artf, xyz):
+        if artf in self.artf:
+            best = None
+            for artf_xyz in self.artf[artf]:
+                if best is None or dist3d(xyz, artf_xyz) < best[0]:
+                    best = dist3d(xyz, artf_xyz), artf_xyz
+            if best is not None and best[0] < 5.0:  # DARPA 5m limit
+                self.artf[artf].remove(best[1])
+                self.score += 1
+                return True
+        return False
+
+
+class MyHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        print("GET:", self.path)
+        s = self.path.split('/')
+        print(s)
```

Comment:
Could all unconditional `print`s be calls to logging?

Reply:
well, I am not sure about `logging` :(. I guess that it has to be configured and we do not have set it up yet. In particular it looks like the HTTP handler has its own logging

Reply:
See https://github.com/robotika/osgar/commit/15a802f9fca49ce2a50794495fed4128edd0113a, if we need to add anything else anywhere, let's add it. For one, it allows us to have one level for record and different, more verbose, level for replay, turn it off for unit tests etc.

Reply:
OK, thanks - fixed & merged